readthedocs/projects/tasks.py

```diff
@@ -404,6 +404,15 @@ def run(self, version_pk):  # pylint: disable=arguments-differ
     default_retry_delay=7 * 60,
 )
 def update_docs_task(self, version_pk, *args, **kwargs):
+    import signal
+
+    def sigterm_received(*args, **kwargs):
+        log.warning('SIGTERM received. Waiting for build to stop gracefully after it finishes.')
+
+    # Do not send the SIGTERM signal to childs
+    # (pip is automatically killed when receives SIGTERM and make the build to fail one command and stop build)
+    signal.signal(signal.SIGTERM, sigterm_received)
```

Comment:
I wonder where this code should live really, in this task seems resonable, but I wonder if there's a celery-level function that would make more sense.

Reply:
Yes. I had the same concern. I was thinking on `readthedocs.worker` but that's executed on normal runs as well.

Reply:
It seems there is a "correct" way of doing this via a Celery plugin. I just found https://github.com/MnogoByte/celery-graceful-stop that replace the SIGTERM handler for the workers as https://github.com/MnogoByte/celery-graceful-stop/blob/master/celery_graceful_stop/bootsteps.py#L35

I will keep reading to see if we can do something similar in a nicer way.